src/components/Tables/DomainTable/DomainTable.tsx

```diff
@@ -175,73 +173,67 @@ const DomainTable: React.FC<DomainTableProps> = ({
 
 	// Gets metadata for each NFT in domain list
 	useEffect(() => {
-		const loaded: DomainData[] = [];
-		setHasMetadataLoaded(false);
-		let count = 0;
-		let completed = 0;
+		let loaded: DomainData[] = [];
 
 		// Get metadata
-		const getData = async (domain: Domain) => {
+		const getData = async (domain: Domain[]) => {
```

Comment:
So it looks like what you've actually done is switch this from fetching all of the domain data in parallel, to instead making it fetch domain data synchronously.

This will actually slow down the total page load time, but make it slightly quicker to show at least one ui element.

Try to take another crack at this. I'm not sure if you're familiar with race conditions, but it's very likely that a race condition could pop up here (when setting `loadedDomains`), in which case you would need to employ a mutex.